stablehlo/dialect/TypeInference.cpp

```diff
@@ -2757,7 +2752,45 @@ LogicalResult inferTriangularSolveOp(
     return emitOptionalError(
         location, "Invalid transpose option value for triangular solve");
 
-  inferredReturnShapes.emplace_back(bType.cast<ShapedType>());
```

Comment:
Given that [the spec](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#triangular_solve) says that "(C4) `b` and `result` have the same type", would it be okay to just inherit bounds from `bType`? That would considerably simplify the resulting code.

Reply:
With further thinking, I agree that we could just infer the bound from `bType`.

Reply:
The code below doesn't seem to be doing that though. Did you mean to keep the `inferMergedDimAndBound` part? If yes, is it intended for it to not apply based on a/b's bounds even though it affects a/b's shapes as well?

Reply:
Thanks, I meant to remove it as well, fixed in the recent commit.